File: CourierAmexAPI/CourierAmex/Models/LocationModel.cs
namespace CourierAmex.Models
{
    public class LocationModel : BaseEntityModel<int>
    {
        public required int CompanyId { get; set; }
        public required string Name { get; set; }
        public required short CountryId { get; set; }
        public string? Phone { get; set; }

        public string? CompanyName { get; set; }
        public string? CountryName { get; set; }
        public bool IsSelected { get; set; }
    }
}

File: CourierAmexClient/src/app/features/company/models/location.model.ts
import { BaseEntity } from "@app/models";

export interface LocationModel extends BaseEntity<number> {
  companyId?: number;
  name: string;
  countryId?: number;
  phone?: string;

  companyName?: string;
  countryName?: string;
  isSelected: boolean;
}

export const newLocationModel = {
  id: 0,
  companyId: undefined,
  countryId: undefined,
  name: '',
  status: 2,
  isSelected: false
};
